src/Razor/src/Microsoft.AspNetCore.Razor.LanguageServer/Semantic/Services/DefaultRazorSemanticTokensInfoService.cs

```diff
@@ -256,30 +288,46 @@ internal class DefaultRazorSemanticTokensInfoService : RazorSemanticTokensInfoSe
             }
 
             // Because SemanticToken data is generated relative to the previous token it must be in order.
-            // We have a guarentee of order within any given language server, but the interweaving of them can be quite complex.
+            // We have a guarantee of order within any given language server, but the interweaving of them can be quite complex.
             // Rather than attempting to reason about transition zones we can simply order our ranges since we know there can be no overlapping range.
             newList.Sort();
 
             return newList;
         }
 
         // Internal and virtual for testing only
-        internal virtual async Task<IReadOnlyList<SemanticRange>?> GetCSharpSemanticRangesAsync(
+        internal virtual async Task<(IReadOnlyList<SemanticRange>? semanticRanges, string? resultId)> GetCSharpSemanticRangesAsync(
```

Comment:
Instead of having the tuples / double args below I'd suggest making a record to represent a versioned semantic range result. Looks like you use these in a few locations so should clean up the usage.